src/core/manager.c

```diff
@@ -1328,6 +1328,14 @@ int manager_startup(Manager *m, FILE *serialization, FDSet *fds) {
         if (r < 0)
                 return r;
 
+        if (m->first_boot && m->unit_file_scope == UNIT_FILE_SYSTEM) {
+                int e = unit_file_preset_all(UNIT_FILE_SYSTEM, 0, NULL, UNIT_FILE_PRESET_ENABLE_ONLY, NULL, 0);
+                if (e < 0)
```

Comment:
We don't prefer reuse of variables, in general, but for stuff like `r` which can be used a hundred times in a function, we do. Either of the proposed options (splitting the declaration, or using `q`) is OK.